.github/workflows/documentation.yml

```diff
@@ -1,29 +1,110 @@
 name: Documentation
+
```

Comment:
I went ahead and added support for deploying docs with DocC. In a later PR I'm going to clean up the README and `Parser` docs to remove information that is covered by the DocC articles.